### PR TITLE
Added support for I3EN_METAL host instance type.

### PR DESCRIPTION
**Pull request by saloneerege**

### CODE REVIEW

File: vmc/resource_vmc_cluster.go [hunk 1]
```diff
@@ -53,12 +53,7 @@ func resourceCluster() *schema.Resource {
 				Optional:    true,
 				Description: "The instance type for the esx hosts added to this cluster.",
 				ValidateFunc: validation.StringInSlice(
-					[]string{HostInstancetypeI3, HostInstancetypeR5}, false),
-			},
-			"storage_capacity": {
```
Comment (shylasrinivas) [inline vmc/resource_vmc_cluster.go:58]: curious to know why storage_capacity is no longer needed

Comment (saloneerege) [reply vmc/resource_vmc_cluster.go:58]: storage capacity is only needed for R5 metal host instance type. R5 is being deprecated moving forward. We will also have to remove support for R5 from cluster and SDDC resource.

Comment (rchadalawada) [reply vmc/resource_vmc_cluster.go:58]: what about customers who have access to R5 hosts? I don't think we should remove it.

Comment (saloneerege) [reply vmc/resource_vmc_cluster.go:58]: We have not tested R5 cluster creation ourselves. We can keep it but we have not tested it ourselves.

File: website/docs/r/sddc.html.markdown [hunk 2]
```diff
@@ -73,7 +73,7 @@ The following arguments are supported:
 
 * `account_link_sddc_config` - (Optional) The account linking configuration object.
 
-* `host_instance_type` -  (Optional) The instance type for the esx hosts in the primary cluster of the SDDC. Possible values : I3_METAL, R5_METAL.
+* `host_instance_type` -  (Optional) The instance type for the esx hosts in the primary cluster of the SDDC. Possible values : I3_METAL, I3EN_METAL, R5_METAL. I3EN_METAL does not currently support single host (1NODE) or two host SDDC deployment.
```
Comment (rchadalawada) [inline website/docs/r/sddc.html.markdown:76]: I don't think this is optional anymore.

File: vmc/constants.go [hunk 3]
```diff
@@ -14,8 +14,9 @@ const (
 	SksNSXTManager string = "/sks-nsxt-manager"
 
 	// ESX Host instance types supported for SDDC creation.
-	HostInstancetypeI3 string = "I3_METAL"
-	HostInstancetypeR5 string = "R5_METAL"
+	HostInstancetypeI3   string = "I3_METAL"
```
Comment (rchadalawada) [inline vmc/constants.go:17]: what happens if we use "i3.metal"?

File: website/docs/r/sddc.html.markdown (hunk 2 above)
Comment (shylasrinivas) [reply website/docs/r/sddc.html.markdown:76]: it has been defined as optional in the schema

File: vmc/constants.go (hunk 3 above)
Comment (lamw) [reply vmc/constants.go:17]: Ideally this matches the VMC API which uses all lower case. Right now, it'll fail the operation and state that its looking for upper case value

**Pull request closed without merging**